src/training/validator.cpp

```diff
@@ -353,6 +355,13 @@ float TranslationValidator::validate(const std::vector<Ptr<ExpressionGraph>>& gr
 
   if(options_->hasAndNotEmpty("valid-translation-output")) {
     fileName = options_->get<std::string>("valid-translation-output");
+    // The formatting below uses fmtlib, which is included with spdlog
+    // and is included via the logger.
+    fileName = fmt::format(fileName.c_str(),
+                           fmt::arg("E", state->epochs),
+                           fmt::arg("U", state->batches),
+                           fmt::arg("B", state->batchesEpoch),
+                           fmt::arg("T", state->labelsTotal));
```

Comment:
Templating is added to TranslationValidator only, but "valid-translation-output" is also used in BleuValidator. We'd like to have this templating in both.

Reply:
Done.